### PR TITLE
Feature/87 restivus replacement

### LINKED ISSUE #68: REST API improvements
Sub-tasks:

* [x] Define authentication method for REST API, token based authentication preferred
* [x] Expose only relevant API endpoints
* [x] Find out how to customize API endpoint routes, would be good to have everything under "/api" -route

### LINKED ISSUE #87: Try Restivus for REST API
Restivus might provide more RESTful approach and better customizability as well as documentation.
Also simple:rest authentication support is pretty much work in progress, https://github.com/stubailo/meteor-rest/issues/95

Try rewriting the Organizations and Departments collections endpoints with Restivus. Bind to existing collections.

### PR DESCRIPTION
**Pull request by jykae**
Closes #87 
Closes #68 

# Proposed changes
- organization and departments CRUD endpoints replaced with Restivus
- enable REST API authentication
- endpoint POST /api/v1/users (Create user)
- GET, DELETE for /api/users/:id (auth required, admin required)
- improved overall configuration (base path, versioning, defaultHeaders, prettyJson, enableCors)

@Digipalvelutehdas/dipor-team please review.

### CODE REVIEW

**Comment by jykae** [comment]
Reminder: Misses all Departments in one Organization endpoint

**Comment by jykae** [comment]
Ready for review @Digipalvelutehdas/dipor-team

**Comment by brylie** [comment]
When renaming template files, be sure to rename the templates themselves. E.g. in `core/client/layouts/navigation.html`, the template should now be named `coreLayoutNavigation`. Update any corresponding routes and template inclusions.

File: core/server/api.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Global API configuration
+ApiV1 = new Restivus({
```
Comment (brylie) [inline core/server/api.js]: Consider using a global `API` object, with attributes for versions. E.g.

```js
API.v1 = {...};
```

As opposed to:
 ``` 
ApiV1 = {};
ApiV2 = {};
ApiV3 = {};
```

File: core/server/api.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  apiPath: 'api/',
+  version: 'v1',
```
Comment (brylie) [inline core/server/api.js]: Is the `v` required here, or is it possible to just use the number `1`?

File: core/server/api.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  defaultHeaders: {
+    'Content-Type': 'application/json'
+  },
+  useDefaultAuth: true,
+  prettyJson: true,
+  enableCors: true
+});
+
+// Generates: POST on /api/users and GET, DELETE /api/users/:id for
```
Comment (brylie) [inline core/server/api.js]: Should this route include the version number?

File: departments/server/api.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ApiV1.addCollection(Departments, {
+  routeOptions: {
+    authRequired: true
+  }
+});
+
+// Maps to: /api/v1/organizations/:id/departments
+ApiV1.addRoute('organizations/:id/departments', {authRequired: true}, {
+  get: function () {
+    const organizationId = this.urlParams.id;
```
Comment (brylie) [inline departments/server/api.js]: Add line comment describing the purpose of this code in plain language.

**Comment by brylie** [comment]
If you want to change the package structure, make sure to change the structure in other project packages. E.g. 

departments/collection/server/publication.js → departments/server/publication.js

Change the `organizations` package to match this new structure, or revert this move.

File: organizations/server/api.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+// Generates: GET, POST on /api/v1/organizations and GET, PUT, DELETE on
+// /api/v1/organizations/:id for the Organizations collection
+ApiV1.addCollection(Organizations, {
+  routeOptions: {
+    authRequired: true
```
Comment (brylie) [inline organizations/server/api.js]: Will this `authRequired` use our collection permissions? E.g. for deleting an organization, what authorization is used?

**Comment by brylie** [comment]
Overall, this looks good. There are just a few questions about the folder/file structure and purpose of some of the code. Inline comments would help to clarify the code for reviewers and new contributors.

**Comment by jykae** [comment]
@brylie Made proposed changes.

**Pull request merged**